src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts

```diff
@@ -144,6 +158,10 @@ export class NfsFormComponent extends CdForm implements OnInit {
         }
       );
 
+      if (this.storageBackend === SUPPORTED_FSAL.RGW) {
+        this.nfsForm.get('rgw_export_type').setValue('bucket');
+        this.resolveBucket();
+      }
       this.getData(promises);
     }
   }
```

Comment:
```
  setRgwExportType(type: RgxExportType) {
    this.nfsForm.get('rgw_export_type').setValue(type);
    switch(type) {
      case RgxExportType.BUCKET:
        this.resolveBucket();
        break;
      case RgxExportType.USER:
        this.resolveUsers();
        break;  
    }
  }
```

```suggestion
  if (this.router.url.startsWith(`/${getPathfromFsal(this.storageBackend)}/nfs/edit`)) {
      this.isEdit = true;
      this.action = this.actionLabels.EDIT;
      this.route.params.subscribe(
        (params: { cluster_id: string; export_id: string; rgw_export_type?: RgxExportType }) => {
          this.cluster_id = decodeURIComponent(params.cluster_id);
          this.export_id = decodeURIComponent(params.export_id);
          this.setRgwExportType(params?.rgw_export_type);
          promises.push(this.nfsService.get(this.cluster_id, this.export_id));
        }
      );
      this.nfsForm.get('cluster_id').disable();
    } else {
      this.action = this.actionLabels.CREATE;
      this.route.params.subscribe(
        (params: { fs_name: string; subvolume_group: string; subvolume?: string }) => {
          this.selectedFsName = params.fs_name;
          this.selectedSubvolGroup = params.subvolume_group;
          if (params.subvolume) this.selectedSubvol = params.subvolume;
        }
      );
      if (this.storageBackend === SUPPORTED_FSAL.RGW) {
        this.setRgwExportType(RgxExportType.BUCKET);;
      }
    }
    this.getData(promises);
```

Reply:
Wel first of all this code of line is used to initialize radio button & defaulted to bucket value! So its not about the switch case here, it must be by default selected as bucket
```
if (this.storageBackend === SUPPORTED_FSAL.RGW) {
        this.nfsForm.get('rgw_export_type').setValue('bucket');
        this.setBucket();
      }
      ````

Reply:
(might be bad choice of function name) My suggestion was to create an initialization function for the buttons only as we  are initiallizing the buttons in both cases - edit and create. in ngOnInit()